include/libopencm3/pac55xx/ccs.h

```diff
@@ -133,6 +218,151 @@ typedef enum {
+/**
+ * Set the divisor for the Auxiliary Clock.
+ * @param[in] div  One of CCS_CTL_CLKDIVx.
+ */
+void ccs_set_aclkdiv(uint32_t div);
+/**
+ * Set the divisor for the AHB Clock.
+ * @param[in] div  One of CCS_CTL_CLKDIVx.
+ */
+void ccs_set_hclkdiv(uint32_t div);
+/** Enable the PLL */
+void ccs_pll_enable(void);
+/** Disable the PLL */
+void ccs_pll_disable(void);
+/** Check if the PLL is locked.
+ * @return true if locked.
+ */
+bool ccs_pll_locked(void);
+/** Enable the PLL bypass */
+void ccs_pll_bypass_enable(void);
+/** Disable the PLL bypass */
+void ccs_pll_bypass_disable(void);
+/**
+ * Set the output divisor.
+ * @param[in] div  Output divisor, one of CCS_PLLCTL_PLLOUTDIVx.
+ */
+void ccs_pll_set_outdiv(uint32_t div);
+/**
+ * Set the PLL input divisor.
+ * @param[in] div  Input divisor, 1-15.
+ */
+void ccs_pll_set_indiv(uint32_t div);
+/**
+ * Set the PLL feedback divisor.
+ * @param[in] div  Feedback divisor, 4-16383.
+ */
+void ccs_pll_set_fbdiv(uint32_t div);
+/**
+ * Configure the CCS PLL, enable it, and wait for lock.
+ * @param[in] indiv  Input divisor, 1-15.
+ * @param[in] fbdiv  Feedback divisor, 4-16383.
+ * @param[in] outdiv  Output divisor, one of CCS_PLLCTL_PLLOUTDIVx.
+ */
+void css_pll_config_enable(uint32_t indiv, uint32_t fbdiv, uint32_t outdiv);
+/**
+ * Get the clock rate (in Hz) of the specified peripheral. This will pull the
+ * proper sources out of the clock tree and calculate the clock for the
+ * peripheral for return to the user, based on current settings.
+ * @param[in] periph  Peripheral base address to get the clock rate for.
+ * @param[in] select  Peripheral-controlled clock select value. Set to 0 when not applicable.
+ * @return Clock rate in Hz for the specified peripheral. 0 if undefined or error.
+ */
+uint32_t ccs_get_peripheral_clk_freq(uint32_t periph, uint32_t select);
+/** Restores CCSCTL and CCSPLLCTL registers to default/safe values */
+void ccs_reset_clocks(void);
+
+/** CCS Clock Configuration structure. */
+struct ccs_clk_config {
+	uint32_t frclk_source;              /**< FRCLK source input selection */
+	uint32_t extclk_frequency;          /**< EXTCLK frequency, 0 if none. */
+	uint32_t sclk_source;               /**< SCLK source selection */
+	uint32_t pll_indiv;                 /**< PLL Input Divider 1-15 */
+	uint32_t pll_fbdiv;                 /**< PLL Feedback Divider 4-16383 */
+	uint32_t pll_outdiv;                /**< PLL Output Divider */
+	uint32_t hclkdiv;                   /**< Divisor from SCLK to HCLK */
+	uint32_t aclkdiv;                   /**< Divisor from SCLK to ACLK */
+	uint32_t pclkdiv;                   /**< Divisor from HCLK to PCLK */
+	uint32_t mem_wstate;                /**< Number of Flash Read wait states */
+	uint32_t mem_mclkdiv;               /**< Divisor from HCLK to MCLK */
+	bool mem_mclksel;                   /**< false: ROSCLK, true: HCLK/MCLK */
+	bool mem_cachedis;                  /**< false: enable cache, true: disable cache */
```

Comment:
I would generally prefer to see positive logic without strong reasons.  (and "default init value is false" isn't really good enough in my mind)

Reply:
yeah, this follows the documentation of the register (below)
![image](https://user-images.githubusercontent.com/37257349/78079479-358f2780-737a-11ea-9fb8-0df5a0b68912.png)

Reply:
and for bitfield definitions, that's more relevant, for user facing helper structures, very much less so.  I see where you're going, but the aims of helpers like this is helping, not just mirroring datasheets.

Reply:
changed as recommended